alerting/src/main/java/org/hawkular/metrics/alerting/PublishCommandTable.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.alerting;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.annotation.Resource;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.hawkular.alerts.api.model.data.CacheKey;
+import org.hawkular.metrics.api.jaxrs.util.Eager;
+import org.hawkular.metrics.model.MetricId;
+import org.hawkular.metrics.model.MetricType;
+import org.infinispan.Cache;
+
+/**
+ * A table to store a cache of published metrics on the bus under hawkular-services context.
+ * Implemented as a shared ISPN cache.
+ *
+ * @author Lucas Ponce
+ */
+@ApplicationScoped
+@Eager
+public class PublishCommandTable {
+
+    private static final String AVAILABILITY = "hm_a_";
+    private static final String COUNTER = "hm_c_";
+    private static final String COUNTER_RATE = "hm_cr_";
+    private static final String GAUGE = "hm_g_";
+    private static final String GAUGE_RATE = "hm_gr_";
+    private static final String STRING = "hm_s_";
+    private static final String UNDEFINED = "hm_u_";
+
+    // This just prevents us from having to create a new object on every cache lookup...
+    // Note, not thread safe, lookups must be atomic
+    Map<MetricType<?>, CacheKey> map = new HashMap<>();
+
+    @Resource(lookup = "java:jboss/infinispan/cache/hawkular-alerts/publish")
+    private Cache<CacheKey, String> publishCache;
+
+    public boolean isPublished(MetricId<?> id) {
+        return publishCache.containsKey(getCacheKey(id));
```

Comment:
Commented on alerts PR but leaving a note here.
Using domain objects in a shared ISPN requires to share classloader.
I think this is adressed in the packaging format.